File: lab3/lab3/1.1/tiger.c
//
//  tiger.c
//  lab3
//
//  Created by Filip Gulan on 11/05/16.
//  Copyright © 2016 Filip Gulan. All rights reserved.
//

#include <stdlib.h>

typedef char const* (*PTRFUN)();

typedef struct {
    PTRFUN *vtable;
    char const* name;
} Tiger;

char const *name(void *this) {
    return ((Tiger *)this)->name;
}

char const *greet() {
    return "Mijau!";
}

char const *menu() {
    return "mlako mlijeko";
}

PTRFUN TigerVTable[3] = {
    (PTRFUN) name,
    (PTRFUN) greet,
    (PTRFUN) menu
};

void initTiger(Tiger *this, char const *name) {
    this->vtable = TigerVTable;
    this->name = name;
}

void *create(char const *name) {
    Tiger *tiger = malloc(sizeof(Tiger));
    initTiger(tiger, name);
    return tiger;
}

File: lab3/1.4/src/hr/fer/zemris/ooup/lab3/zad_1_4/model/plugins/Tiger.java
package hr.fer.zemris.ooup.lab3.zad_1_4.model.plugins;

import hr.fer.zemris.ooup.lab3.zad_1_4.model.Animal;

/**
 * @author Filip Gulan
 */
public class Tiger extends Animal {

    private String name;

    public Tiger(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }

    @Override
    public String greet() {
        return "Mijau!";
    }

    @Override
    public String menu() {
        return "mlako mlijeko";
    }
}
